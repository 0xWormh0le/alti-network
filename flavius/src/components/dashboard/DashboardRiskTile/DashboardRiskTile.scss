@import 'styles/_shared.scss';

.DashboardRiskTile {
  border-radius: 6px;
  border: solid 1px $PALE_GREY;
  min-height: 0;
  height: 100%;
  flex: 1 1 0;
  padding: $SMALL_SPACING;
  @include flex(column, space-between, stretch);

  a {
    text-decoration: none;
  }

  &__button {
    border: none;
    border-radius: 5px;
    background-color: $PALE_GREY_LIGHTEST;
    display: flex;
    width: 100%;
    min-height: 0;
    padding: $SMALL_SPACING / 2 $SMALL_SPACING;
    flex: 1;
    margin-top: $BASE_SPACING;
    @include flex(row, center, center);
    transition: background-color 0.2s ease-in-out;

    &:hover,
    &:focus {
      background-color: $LIGHT_BLUE;
    }

    &:active {
      outline: none;
    }
  }

  &__title {
    border-bottom: solid 1px $PALE_GREY;
    padding-bottom: 17px;
    margin-bottom: 19px;
    font-weight: 900;
  }

  &__label {
    color: $LIGHT_GREY;
    max-width: 60%;
    @include flex(column, center, center);
  }

  &__count {
    color: $DARK_GREY_BLUE;
    font-weight: 600;
    margin-right: $SMALL_SPACING / 2;
  }
}
