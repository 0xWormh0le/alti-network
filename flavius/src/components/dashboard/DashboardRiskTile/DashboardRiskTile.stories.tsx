import React from 'react'
import { Story, Meta } from '@storybook/react/types-6-0'
import DashboardRiskTile, { DashboardRiskTileProps } from './DashboardRiskTile'
import { BrowserRouter } from 'react-router-dom'

export default {
  title: 'Dashboard/DashboardRiskTile',
  component: DashboardRiskTile
} as Meta

const Template: Story<DashboardRiskTileProps> = (args) => (
  <BrowserRouter>
    <DashboardRiskTile {...args} />
  </BrowserRouter>
)

export const Default = Template.bind({})
Default.args = {
  title: 'Title',
  count: 1,
  label: 'Label'
}
