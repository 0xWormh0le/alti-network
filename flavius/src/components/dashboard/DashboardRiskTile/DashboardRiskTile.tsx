import React from 'react'

import Typography, { TypographyVariant } from 'components/elements/Typography'
import { Link } from 'react-router-dom'
import './DashboardRiskTile.scss'

export interface DashboardRiskTileProps {
  count: number
  label: string
  linkTo: string
  title: string
  children: React.ReactNode
}

export const DashboardRiskTile: React.FC<DashboardRiskTileProps> = ({ count, label, linkTo, title, children }) => {
  const { H1, LABEL } = TypographyVariant

  return (
    <div className='DashboardRiskTile'>
      <Typography variant={LABEL} component='div' className='DashboardRiskTile__title'>
        {title}
      </Typography>
      {children}
      <Link to={linkTo}>
        <button className='DashboardRiskTile__button'>
          <Typography variant={H1} component='span' className='DashboardRiskTile__count'>
            {count}
          </Typography>
          <Typography variant={LABEL} component='span' className='DashboardRiskTile__label'>
            {label}
          </Typography>
        </button>
      </Link>
    </div>
  )
}

export default DashboardRiskTile
