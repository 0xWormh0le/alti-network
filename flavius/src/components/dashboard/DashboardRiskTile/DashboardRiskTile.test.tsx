import React from 'react'
import { render } from '@testing-library/react'
import { DashboardRiskTile, DashboardRiskTileProps } from './DashboardRiskTile'
import { BrowserRouter } from 'react-router-dom'

describe('DashboardRiskTile', () => {
  it('renders correctly', () => {
    const props: DashboardRiskTileProps = {
      count: 1000,
      label: 'risks associated',
      linkTo: '/',
      children: 'children test',
      title: 'title test'
    }

    // needs router since it contains a link
    const { container } = render(
      <BrowserRouter>
        <DashboardRiskTile {...props} />
      </BrowserRouter>
    )
    expect(container).toMatchSnapshot()
  })
})
