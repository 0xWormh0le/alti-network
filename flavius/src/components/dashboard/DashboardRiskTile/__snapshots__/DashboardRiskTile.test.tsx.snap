// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DashboardRiskTile renders correctly 1`] = `
<div>
  <div
    class="DashboardRiskTile"
  >
    <div
      class="Typography Typography--label DashboardRiskTile__title"
    >
      title test
    </div>
    children test
    <a
      href="/"
    >
      <button
        class="DashboardRiskTile__button"
      >
        <span
          class="Typography Typography--h1 DashboardRiskTile__count"
        >
          1000
        </span>
        <span
          class="Typography Typography--label DashboardRiskTile__label"
        >
          risks associated
        </span>
      </button>
    </a>
  </div>
</div>
`;
